src/regdesk/printer_manager.js

```diff
@@ -0,0 +1,66 @@
+import { LabelEpl, LP2844 } from 'WebZLP/src/LP2844'
+
+export class PrinterManager {
+    #devices = [];
+    #printers = [];
+
+    primaryPrinter = 0;
+    secondaryPrinter = 1;
+
+    #printerType = LP2844;
+
+    constructor() {
+    }
+
+    get printers() { return this.#printers; }
+
+    async printerConnect(e) {
+        if (!this.#devices.includes(e.device)) {
```

Comment:
I'm a fan of de-nesting if statements. So that this would become

```
if(this.#devices.includes(e.device) {
  return;
}
this.#devices.push(e.device);
...
```

https://softwareengineering.stackexchange.com/questions/47789/how-would-you-refactor-nested-if-statements

Reply:
Cleaned up a bit. The original method here grew very organically over time.